013-kafka-topic-encryption.md

```diff
@@ -4,38 +4,93 @@
+
+#### Encrypter-Decrypter
+The Encrypter-Decrypter sub-component performs the actual encryption and decryption of messages, implementing a particular encryption algorithm and making use of the KMS service to obtain topic keys.
+
+#### Key Service
+The Key Service is used in the Encryption Module to handle interactions with a key management system and to perform optional primitives such as those related to envelope encryption. The interface is generic in order to accommodate various key management implementations.
+
+#### Policy Service
+Encrypter-decrypters require a means to lookup encryption requirements and parameters for a topic. As a corresponding standard does not exist, the core component will interact with a generic internal interface for retrieving policy on how a topic is, or not is, to be encrypted. Implementers can develop and configure their own implementation of the policy service, be it a remote microservice, a local pre-configured cache, a cluster registry, etc.
+
+For each topic to be encrypted, a policy exists detailing:
+
+- topic name
+- encryption algorithm(s), cipher suite,  and encryption/decryption steps
+- key management server address
+- key identifier
+- optional initialization information
+
+
+### Message Interception
+Message interception is concerned with facilities for inspecting messages, consulting a policy, and accordingly applying encryption so that messages are passed to the broker in encrypted form and returned to Kafka clients as decrypted plaintext. Specifically, Kafka Produce requests and Fetch responses are examined and potentially modified to respectively encrypt and decrypt messages.
+
+### Proxy
+A proxy can be implemented as a free-standing process or as a sidecar. In both cases, the proxy needs to
+intercept and de-serialize the Kafka messages, encrypt/decrypt the relevant parts, 
+and reserialize them. The overhead of a proxy potentially has an effect on performance.
+It also means that the proxy version must always be in phase with that of the broker.
+A proxy-based solution however has the advantage that both Kafka client and broker are unaware
+of the proxy and do not require any modification or configuration to support encryption at rest.
+
+Initially a free-standing proxy will be developed. 
+
+A second proxy option is Envoy. Envoy is a general framework for creating proxies within Kubernetes used for example by Istio. Envoy’s connection is based on network filters that can be linked into filter chains, allowing rich capabilities to be composed from small microservices. This is contrast to monolithic proxy solutions like NGINX. Envoy's fairly recent support of WebAssembly (WASM) is the preferred model for introducing the topic encryption into an Envoy proxy.
+
+
 
 ## Affected/not affected projects
 
 Call out the projects in the Strimzi organisation that are/are not affected by this proposal. 
-here comes text
+
 
 ## Compatibility
```

Comment:
Should we add here that the proxy needs to be compatible to future message formats of the producerequest and fetchresponse?